packages/client-sdk-nodejs/src/internal/cache-control-client.ts

```diff
@@ -113,15 +122,20 @@ export class CacheControlClient {
       cache_name: name,
     });
     this.logger.debug(`Deleting cache: ${name}`);
-    return await new Promise<DeleteCache.Response>(resolve => {
+    return await new Promise<DeleteCache.Response>((resolve, reject) => {
       this.clientWrapper
         .getClient()
         .DeleteCache(
           request,
           {interceptors: this.interceptors},
           (err, _resp) => {
             if (err) {
-              resolve(new DeleteCache.Error(cacheServiceErrorMapper(err)));
+              this.cacheServiceErrorMapper.handleError(
+                err,
+                e => new DeleteCache.Error(e),
+                resolve,
```

Comment:
is there a compilation failure if you accidentally swap the resolve and reject functions?

Reply:
I don't think so. Presumably you would get some int test failures though because nothing would succeed?

Also open to ideas on how to improve this.

Reply:
the idea that comes to mind first is having the function take in an object with named arguments, but I don't think that its a dealbreaker,  and that would require changing basically every line in this pr 😅 . I think how it currently is is good

Reply:
this is done